File: data_structures/programmers_lotto_max_min_rank.cpp
/*
테스트 1 〉	통과 (0.01ms, 3.66MB)
테스트 2 〉	통과 (0.01ms, 4.16MB)
테스트 3 〉	통과 (0.01ms, 4.18MB)
테스트 4 〉	통과 (0.01ms, 4.17MB)
테스트 5 〉	통과 (0.01ms, 4.17MB)
테스트 6 〉	통과 (0.01ms, 4.14MB)
테스트 7 〉	통과 (0.01ms, 4.09MB)
테스트 8 〉	통과 (0.01ms, 4.25MB)
테스트 9 〉	통과 (0.01ms, 4.2MB)
테스트 10 〉	통과 (0.01ms, 4.16MB)
테스트 11 〉	통과 (0.01ms, 4.17MB)
테스트 12 〉	통과 (0.01ms, 4.18MB)
테스트 13 〉	통과 (0.01ms, 4.09MB)
테스트 14 〉	통과 (0.01ms, 3.67MB)
테스트 15 〉	통과 (0.01ms, 4.03MB)
*/

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

// 알수 없는 번호 = 0
// 최고 순위와 최저 순위 알아보기
// 최고 순위, 최저 순위 리턴

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    
    int cnt_match = 0;
    int cnt_zero = 0;
    
    for (auto& l : lottos) {
        if (l == 0) cnt_zero++;
        else {
            if (find(win_nums.begin(), win_nums.end(), l) != win_nums.end()) cnt_match++;
        }
    }
    
    // 최고 순위
    int max_match = cnt_match + cnt_zero;
    if (max_match <= 1) answer.push_back(6);
    else answer.push_back(7-max_match);
    
    // 최저 순위
    if (cnt_match <= 1) answer.push_back(6);
    else answer.push_back(7-cnt_match);
    
    return answer;
}

File: data_structures/programmers_lotto_max_min_rank.java
/*
테스트 1 〉	통과 (0.04ms, 88.1MB)
테스트 2 〉	통과 (0.05ms, 75.4MB)
테스트 3 〉	통과 (0.05ms, 76.4MB)
테스트 4 〉	통과 (0.05ms, 84.2MB)
테스트 5 〉	통과 (0.04ms, 77.3MB)
테스트 6 〉	통과 (0.04ms, 73.6MB)
테스트 7 〉	통과 (0.05ms, 78.7MB)
테스트 8 〉	통과 (0.04ms, 74.3MB)
테스트 9 〉	통과 (0.04ms, 84.5MB)
테스트 10 〉	통과 (0.04ms, 77.3MB)
테스트 11 〉	통과 (0.04ms, 74.2MB)
테스트 12 〉	통과 (0.05ms, 75.7MB)
테스트 13 〉	통과 (0.05ms, 72.1MB)
테스트 14 〉	통과 (0.05ms, 74.3MB)
테스트 15 〉	통과 (0.04ms, 73.9MB)
*/

import java.util.*;
// 복권 최고, 최저 순위

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int cnt = 0;
        int cnt_zero = 0;
        
        List<Integer> nums = new ArrayList<Integer>();
        for (int n : win_nums) nums.add(n);
        
        for (int l : lottos) {
            if (nums.contains(l)) cnt++;
            if (l == 0) cnt_zero++;
        }
        
        int[] answer = new int[2];
        answer[0] = (cnt+cnt_zero < 2) ? 6 : (7-(cnt+cnt_zero));
        answer[1] = (cnt < 2) ? 6 : (7-cnt);
        
        return answer;
    }
}
